Updates/0.3.3.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\QueuedTracking;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\QueuedTracking\Queue\Manager;
+use Piwik\Updater;
+use Piwik\Updates;
+
+class Updates_0_3_3 extends Updates
+{
+    /**
+     * @var Manager
+     */
+    protected $queueManager;
+
+    public function doUpdate(Updater $updater)
+    {
+        $settings = $this->getSettings();
+        $numQueueWorkers = $settings->numQueueWorkers->getValue();
+
+        if ($numQueueWorkers <= 1) {
+            return; // it is pointless to rewrite one queue to the same queue
+        }
+
+        $this->rewriteQueues($numQueueWorkers);
```

Comment:
Maybe, the trick could be to write the requests one at a time into different queues and then later rename them or move them into the right queue.

Eg if there are supposed to be 16 queues, you could write them first into queue 17-32 or into `prefix_queue_1...16` one or X requests at a time. Then once you are through all requests rename eg `queue 17` to `queue 1` or `prefix_queue_1` to `queue_1`. Or maybe if rename does not work simply move all requests from eg queue 17 to queue 1 and delete queue 17 afterwards. 

Just a thought. If it was possible to rename a queue eg from 17 to 1, it would be best. If it takes eg 3 seconds to redistribute requests, by the time you finished redistributing the requests there might be meanwhile new requests for queue 1. But then we could simply discard the requests from last 3 seconds my moving queue 17 to 1 and everything would be still tracked correctly apart from 3 seconds.

This logic of moving them first to another queue would also help for `moveRequestsIntoAnotherQueue()` and we could remove `moveRequestsIntoAnotherQueue ()` in favor of that new method